tests/PHPSA/Expression/Operators/Logical/BooleanAndTest.php

```diff
@@ -18,6 +18,16 @@ public function getDataProvider()
             array(false, true, false),
             array(true, false, false),
             array(false, false, false),
+            array(null, null, false),
+            array(true, null, false),
+            array(null, true, false),
+            array(1, true, true),
+            array(1.4, true, true),
+            array(1, false, false),
+            array(-1, true, true),
+            array("a", true, true),
+            array(array(), array(), false),
+            array(array(), "a", false),
```

Comment:
The method newFakeScalar which we are using is not yet supporting arrays that are not empty or objects. So we are testing those for nothing at the moment.
We will have to add those to fakescalar (but they are not real scalars so maybe rename this) or make a new testcase (but why should we make 2 testcases when we are testing the same thing basically) So I would prefer the first but this would be more something for another PR (pretty much work)